waspc/cli/src/Wasp/Cli/Command/Db/Start.hs

```diff
@@ -0,0 +1,63 @@
+{-# LANGUAGE TypeApplications #-}
+
+module Wasp.Cli.Command.Db.Start
+  ( start,
+  )
+where
+
+import Control.Monad.IO.Class (liftIO)
+import StrongPath (Abs, Dir, Dir', Path', Rel, fromAbsDir, reldir, (</>))
+import System.Process (callCommand)
+import qualified Wasp.AppSpec as AS
+import qualified Wasp.AppSpec.App as AS.App
+import qualified Wasp.AppSpec.App.Db as AS.App.Db
+import Wasp.Cli.Command (Command)
+import Wasp.Cli.Command.Common
+  ( findWaspProjectRootDirFromCwd,
+    parseWaspFileC,
+  )
+import Wasp.Cli.Command.Message (cliSendMessageC)
+import Wasp.Cli.Common (WaspProjectDir, dotWaspDirInWaspProjectDir)
+import qualified Wasp.Db.Postgres as P
+import qualified Wasp.Message as Msg
+
+start :: Command ()
+start = do
+  waspProjectDir <- findWaspProjectRootDirFromCwd
+  decls <- parseWaspFileC waspProjectDir
+
+  case getDbSystem decls of
+    Nothing -> noteSQLiteDoesntNeedStart
+    Just AS.App.Db.SQLite -> noteSQLiteDoesntNeedStart
+    Just AS.App.Db.PostgreSQL -> startPostgreDevDb waspProjectDir
+  where
+    getDbSystem decls = (snd <$> AS.getApp decls) >>= AS.App.db >>= AS.App.Db.system
+
+    noteSQLiteDoesntNeedStart =
+      cliSendMessageC . Msg.Info $
+        "Nothing to do! You are all good, you are using SQLite which doesn't need to be started."
+
+startPostgreDevDb :: Path' Abs (Dir WaspProjectDir) -> Command ()
+startPostgreDevDb waspProjectDir = do
+  let devDbDir = waspProjectDir </> devDbDataDirInWaspProjectDir
+  -- NOTE: POSTGRES_PASSWORD, POSTGRES_USER, POSTGRES_DB below are really used by the docker image
+  --   only when initializing the database -> if it already exists, they will be ignored.
+  let command =
+        unwords
+          [ "docker run",
+            "--rm",
+            "--publish " <> show P.defaultDevPort <> ":5432",
+            "-v " <> fromAbsDir devDbDir <> ":/var/lib/postgresql/data",
+            "--env POSTGRES_PASSWORD=" <> P.defaultDevPass,
+            "--env POSTGRES_USER=" <> P.defaultDevUser,
+            "--env POSTGRES_DB=" <> P.defaultDevDbName,
+            "postgres"
+          ]
+  liftIO $ callCommand command
```

Comment:
Maybe we can run `migrate-dev` after we start the database? 🤷 I'm not sure how much this complicates things.

Reply:
That would be cool! I had that down as an idea, but decided to punt on it for now, I think it could potentially complicate things quite a bit right now.

Reply:
Gave up on this but came up with a better idea that is more general, will create an issue for it (it is under TODOs in the main comment of this PR).